src/invocation/ClusterService.ts

```diff
@@ -14,317 +14,283 @@
  * limitations under the License.
  */
 
-import {ClientConnection} from './ClientConnection';
+import {ClientConnection} from '../network/ClientConnection';
 import * as Promise from 'bluebird';
-import {ClientAddMembershipListenerCodec} from '../codec/ClientAddMembershipListenerCodec';
 import {Member} from '../core/Member';
-import {LoggingService} from '../logging/LoggingService';
 import {ClientInfo} from '../ClientInfo';
 import HazelcastClient from '../HazelcastClient';
-import {IllegalStateError} from '../HazelcastError';
-import * as assert from 'assert';
+import {IllegalStateError, TargetDisconnectedError} from '../HazelcastError';
 import {MemberSelector} from '../core/MemberSelector';
-import {AddressHelper, DeferredPromise} from '../Util';
-import {MemberAttributeEvent, MemberAttributeOperationType} from '../core/MemberAttributeEvent';
+import {assertNotNull, DeferredPromise} from '../Util';
 import {MembershipListener} from '../core/MembershipListener';
 import {MembershipEvent} from '../core/MembershipEvent';
 import {UuidUtil} from '../util/UuidUtil';
 import {ILogger} from '../logging/ILogger';
-import Address = require('../Address');
-import ClientMessage = require('../ClientMessage');
+import {UUID} from '../core/UUID';
+import {ClientConnectionManager} from '../network/ClientConnectionManager';
+import {InitialMembershipListener} from '../core/InitialMembershipListener';
+import {InitialMembershipEvent} from '../core/InitialMembershipEvent';
+import {MemberInfo} from '../core/MemberInfo';
+import {Cluster} from '../core/Cluster';
 
 export enum MemberEvent {
     ADDED = 1,
     REMOVED = 2,
 }
 
-/**
- * Manages the relationship of this client with the cluster.
- */
-export class ClusterService {
+class MemberListSnapshot {
+    version: number;
```

Comment:
nit: fields of this class could be marked as `readonly`.

Reply:
Version is not readonly, we reset it in `clearMemberListVersion` method. But I made the other fields readonly

Reply:
My bad. Thanks for changing modifiers for other fields.